pandas/core/ops.py

```diff
@@ -657,17 +658,46 @@ def wrapper(self, other):
         if isinstance(other, pd.Series):
             name = _maybe_match_name(self, other)
 
-            other = other.reindex_like(self).fillna(False).astype(bool)
-            return self._constructor(na_op(self.values, other.values),
+            other = other.reindex_like(self)
+            if other.dtype.kind not in ['u', 'i']:
```

Comment:
look in ``core.common`` for function like ``is_interger_dtype`` and such. Pls use these rather then checking dtypes like this